src/rpc/blockchain.cpp

```diff
@@ -518,7 +518,9 @@ static UniValue getrawmempool(const JSONRPCRequest& request)
     if (!request.params[0].isNull())
         fVerbose = request.params[0].get_bool();
 
-    return MempoolToJSON(::mempool, fVerbose);
+    const CTxMemPool* tx_pool = MempoolInstance();
+    CHECK_NONFATAL(tx_pool);
```

Comment:
Due to RPC requests only being served after `SetRPCWarmupFinished` even after mempool is switched to a struct initialized during init sequence, it shouldn't never be null (but better to be secure if we refactor our init logic)

Reply:
Yes, obviously this should be adjusted when the mempool is truly optional. However I don't see the reason to add dead code right now.

Reply:
My point was more currently the `CHECK_FATAL` is useless because RPC server accepting requests is the last thing in our init sequence. `CHECK_FATAL` hitting  would be a glaring logic error.

Reply:
> CHECK_FATAL is useless

CHECK_NONFATAL is not useless. Its exact use case it to be used when the condition is assumed to be always true. This is the case here. See https://dev.visucore.com/bitcoin/doxygen/check_8h.html#a46a3e27097aa5e94bbf62075bad7016f

I plan to introduce a `-nomempool` command line option, which makes the mempool go away (optional). When I fail to update this line of code, it will turn into a logic error and will properly throw a bug report.